xrpl/models/transactions/xchain_claim.py

```diff
@@ -77,7 +78,17 @@ def _get_errors(self: Self) -> Dict[str, str]:
         errors = super()._get_errors()
 
         bridge = self.xchain_bridge
-        currency = XRP() if isinstance(self.amount, str) else self.amount.to_currency()
+
+        amount = self.amount
+        if is_xrp(amount):
+            currency = XRP()
+        elif is_issued_currency(amount):
+            currency = amount.to_currency()  # type: ignore
+        elif is_mpt(amount):
+            currency = amount.mpt_issuance_id  # type: ignore
+        else:
+            errors["amount"] = "currency can't be derived."
+
         if (
             currency != bridge.locking_chain_issue
             and currency != bridge.issuing_chain_issue
```

Comment:
Seems like this last error isn't tracked in the JS lib. Is this no longer relevant or should it be added to the JS library as well?